lark/parser_frontends.py

```diff
@@ -27,56 +26,112 @@ def lex(self, lexer_state, parser_state):
+            self.skip_lexer = True
+            return
+
+        try:
+            create_lexer = {
+                'standard': create_traditional_lexer,
+                'contextual': create_contextual_lexer,
+            }[lexer_conf.name]
+        except KeyError:
+            assert issubclass(lexer_conf.name, Lexer), lexer_conf.name
+            self.lexer = _wrap_lexer(lexer_conf.name)(lexer_conf)
+        else:
+            self.lexer = create_lexer(lexer_conf, self.parser, lexer_conf.postlex)
+
+        if lexer_conf.postlex:
+            self.lexer = PostLexConnector(self.lexer, lexer_conf.postlex)
+
+
+    def _parse(self, start, input, *args):
+        if start is None:
+            start = self.parser_conf.start
+            if len(start) > 1:
+                raise ConfigurationError("Lark initialized with more than 1 possible start rule. Must specify which start rule to parse", start)
+            start ,= start
+        return self.parser.parse(input, start, *args)
+
+    def parse(self, text, start=None):
+        if self.skip_lexer:
+            return self._parse(start, text)
+
+        lexer = LexerThread(self.lexer, text)
+        return self._parse(start, lexer)
+
+
 def get_frontend(parser, lexer):
     if parser=='lalr':
         if lexer is None:
             raise ConfigurationError('The LALR parser requires use of a lexer')
-        elif lexer == 'standard':
-            return LALR_TraditionalLexer
-        elif lexer == 'contextual':
-            return LALR_ContextualLexer
-        elif issubclass(lexer, Lexer):
-            wrapped = _wrap_lexer(lexer)
-            class LALR_CustomLexerWrapper(LALR_WithLexer):
-                def init_lexer(self):
-                    self.lexer = wrapped(self.lexer_conf)
-            return LALR_CustomLexerWrapper
-        else:
+        if lexer not in ('standard' ,'contextual') and not issubclass(lexer, Lexer):
             raise ConfigurationError('Unknown lexer: %s' % lexer)
     elif parser=='earley':
-        if lexer=='standard':
-            return Earley_Traditional
-        elif lexer=='dynamic':
-            return XEarley
-        elif lexer=='dynamic_complete':
-            return XEarley_CompleteLex
-        elif lexer=='contextual':
+        if lexer=='contextual':
             raise ConfigurationError('The Earley parser does not support the contextual parser')
-        elif issubclass(lexer, Lexer):
-            wrapped = _wrap_lexer(lexer)
-            class Earley_CustomLexerWrapper(Earley_WithLexer):
-                def init_lexer(self, **kw):
-                    self.lexer = wrapped(self.lexer_conf)
-            return Earley_CustomLexerWrapper
-        else:
-            raise ConfigurationError('Unknown lexer: %s' % lexer)
     elif parser == 'cyk':
-        if lexer == 'standard':
-            return CYK
-        else:
+        if lexer != 'standard':
             raise ConfigurationError('CYK parser requires using standard parser.')
```

Comment:
Could Cyk be made to support Custom Lexer? (I mean, I don't know why, but it should work, shouldn't it)

Reply:
Yeah, it should work. Maybe it already works and we just need to remove this check.

Reply:
Yep. Easy.